### PR TITLE
iOS/Safari support, resolveLocalFileSystemURL, allow debugging in Chrome

### PR DESCRIPTION
**Pull request by mudcube**

### CODE REVIEW

File: src/idb.filesystem.js [hunk 1]
```diff
@@ -37,13 +37,63 @@ if (exports.requestFileSystem || exports.webkitRequestFileSystem) {
...
+      }
+    };
+    request.onupgradeneeded = function() {
+      request.result.createObjectStore("store");
+    };
+  };
+};
+
+var Base64ToBlob = function(uri) {
+  uri = uri.split(',');
```
Comment (ebidel) [inline src/idb.filesystem.js]: can we use this impl: https://github.com/ebidel/filer.js/blob/master/src/filer.js#L137. It's well tested :)

File: src/idb.filesystem.js [hunk 2]
```diff
@@ -37,13 +37,63 @@ if (exports.requestFileSystem || exports.webkitRequestFileSystem) {
...
+  var dbName = "blob-support";
+  indexedDB.deleteDatabase(dbName).onsuccess = function() {
+    var request = indexedDB.open(dbName, 1.0);
+    request.onerror = function() {
+      support.blob = false;
+    };
+    request.onsuccess = function() {
+      var db = request.result;
+      try {
+        var blob = new Blob(["test"], {type:"text/plain"});
```
Comment (ebidel) [inline src/idb.filesystem.js]: `type: "text/plain`

File: src/idb.filesystem.js [hunk 3]
```diff
@@ -37,13 +37,63 @@ if (exports.requestFileSystem || exports.webkitRequestFileSystem) {
...
+  var uint = new Uint8Array(buffer);
+  for (var n = 0; n < binary.length; n++) {
+    uint[n] = binary.charCodeAt(n);
+  }
+  var blob = new Blob([buffer], {type: mime});
+  return blob;
+};
+
+var BlobToBase64 = function(blob, onload) {
+  var reader = new window.FileReader();
```
Comment (ebidel) [inline src/idb.filesystem.js]: remove `window.`

File: src/idb.filesystem.js [hunk 4]
```diff
@@ -290,21 +340,33 @@ function FileWriter(fileEntry) {
...
 
-    // Set the blob we're writing on this file entry so we can recall it later.
-    fileEntry.file_.blob_ = blob_;
-    //fileEntry.file_.blob_.lastModifiedDate = data.lastModifiedDate || null;
-    fileEntry.file_.lastModifiedDate = data.lastModifiedDate || null;
+    var writeFile = function(blob) {
+      // Blob might be a DataURI depending on browser support
+      fileEntry.file_.blob_ = blob;
+      // fileEntry.file_.blob_.lastModifiedDate = data.lastModifiedDate || null;
+      fileEntry.file_.lastModifiedDate = data.lastModifiedDate || new Date();
```
Comment (ebidel) [inline src/idb.filesystem.js]: Not sure how I feel about adding the artificial `new Date()`

File: src/idb.filesystem.js [hunk 5]
```diff
@@ -711,11 +776,22 @@ function requestFileSystem(type, size, successCallback, opt_errorCallback) {
...
-function resolveLocalFileSystemURL(url, callback, opt_errorCallback) {
-  if (opt_errorCallback) {
-    opt_errorCallback(NOT_IMPLEMENTED_ERR);
-    return;
-  }
+function resolveLocalFileSystemURL(url, successCallback, opt_errorCallback) {
+  var origin = location.protocol + '//' + location.host;
+  var base = 'filesystem:' + origin + DIR_SEPARATOR + storageType_.toLowerCase();
+  url = url.replace(base, '');
+  if (url.substr(-1) === '/') url = url.slice(0, -1);
```
Comment (ebidel) [inline src/idb.filesystem.js]: brackets please

**Comment by ebidel** [comment]
Tests please :)

**Comment by mudcube** [comment]
No problemo. On the requested tests, were you thinking about for 'resolveLocalFileSystemURL' or another thing in mind as well? I haven't messed with quint before, so will take a look soon.

**Comment by ebidel** [comment]
resolveLocalFileSystemURL, the Base* methods, and support methods if you can.

**Comment by mudcube** [comment]
Ok, I added some tests for resolveLocalFileSystemURL, and the Base64 test from Filer.js.  I'm not sure how to test whether the support method works properly, because it's the test itself--thoughts on that?

**Comment by mudcube** [comment]
Oh, I removed the new Date() in effort to address what you mentioned on Jan 13th:

"Not sure how I feel about adding the artificial new Date()"

Do you think it'd be best to keep the artificial date? That'd be my preference.

**Comment by ebidel** [comment]
Doh. Sorry about that. I was only reviewing the latest commits. That's my bad for not getting to the PR sooner.

LGTM.

**Pull request merged**